types/src/main/scala/aqua/types/Type.scala

```diff
@@ -12,12 +12,96 @@ sealed trait Type {
     import cats.syntax.partialOrder._
     this >= incoming
   }
+
+  def isInhabited: Boolean = true
+}
+
+// Product is an ordered list of types, optionally labelled
```

Comment:
what does it mean `ordered` and `labelled`? more description in docs to make it clear what it is

Reply:
add that `ordered` is for equality between types